### PR TITLE
Add 3 new list items: card, item large and item small

### PR DESCRIPTION
**Pull request by MichelGabriel**
This PR adds 3 new list components: 
- card
- item large
- item small

and introduce new class: `.clamp-x`

gn-ui-list-components

### CODE REVIEW

File: libs/ui/src/lib/record-preview-list/record-preview-list.component.html [hunk 1]
```diff
@@ -1,17 +1,155 @@
...
+        {{ record.title }}
+      </h1>
+      <p class="leading-relaxed mb-3 clamp-3">{{ record.abstract }}</p>
+
+      <div class="hidden sm:flex-1 sm:flex sm:items-center sm:justify-between">
+        <a
+          class="inline-flex items-center md:mb-2 lg:mb-0 hover:underline"
+          [href]="record.url"
+        >
+          {{ record.more }}
```
Comment (fgravin) [inline libs/ui/src/lib/record-preview-list/record-preview-list.component.html:27]: This should be a translation, it's not a field in metadata record
```html
<span translate>readMore</span>
```

File: libs/search/src/lib/state/effects.ts [hunk 2]
```diff
@@ -65,6 +65,15 @@ export class SearchEffects {
             : 'no abstract',
           thumbnailUrl: hit._source.overview ? hit._source.overview.url : '',
           url: `/geonetwork/srv/eng/catalog.search#/metadata/${hit._source.uuid}`,
+          category: hit._source.resourceCategoryObject
```
Comment (fgravin) [inline libs/search/src/lib/state/effects.ts:68]: You misread the purpose of this mapping.
This is a mapper between ElasticSearch index and `RecordSimple` interface. `resourceCategoryObject`, `resourceMoreObject` and `resourceOpenObject` don't exist in ES search response.

File: libs/search/src/lib/model.ts [hunk 3]
```diff
@@ -7,6 +7,9 @@ export interface RecordSimple {
   abstract: string
   url: string
   thumbnailUrl: string
+  category: string
+  more: string
```
Comment (fgravin) [inline libs/search/src/lib/model.ts:11]: What is `more`, what is `open` ?
These are properties of the metadata itself, there is a miss understanding here.

**Review by fgravin — CHANGES_REQUESTED** [review]
The HTML look fantastic but there is a misunderstanding about the RecordSimple interface and ES mappers.

File: libs/search/src/lib/state/effects.ts (hunk 2 above)
Comment (MichelGabriel) [reply libs/search/src/lib/state/effects.ts:68]: Ok, I will remove them

File: libs/search/src/lib/model.ts (hunk 3 above)
Comment (MichelGabriel) [reply libs/search/src/lib/model.ts]: I removed those, they should be gone

**Comment by MichelGabriel** [comment]
> don't forget to update the template accordingly
> I think you still want the metadata category don't you ?

Yes, category should be there.

**Comment by fxprunayre** [comment]
> Yes, category should be there.

What do you mean by category? The db category ? Usually we prefer to use ISO topics or a thesaurus like INSPIRE themes that can be represented using icons.

**Comment by MichelGabriel** [comment]
> > Yes, category should be there.
> 
> What do you mean by category? The db category ? Usually we prefer to use ISO topics or a thesaurus like INSPIRE themes that can be represented using icons.

I'm going to remove it altogether, it's not really needed in the list items

File: libs/ui/src/lib/record-preview-list/record-preview-list.stories.ts [hunk 4]
```diff
@@ -20,10 +20,14 @@ RecordPreviewListComponentStory.args = {
...
     uuid: 'uiiudiiddeaafdjsqmlkfdq',
     title: 'A very nice record',
+    category: 'My Category',
     thumbnailUrl:
       'https://sextant.ifremer.fr/var/storage/images/_aliases/listitem_thumbnail/medias-ifremer/medias-sextant/accueil/cartes-thematiques/adcp/1595636-3-fre-FR/ADCP.png',
-    abstract: 'this is a great abstract',
+    abstract:
+      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam ultrices turpis nec eleifend ornare. Etiam ante est, pulvinar ac diam nec, sollicitudin tristique leo. Vivamus a dui sed ex mattis tempus non in urna. Phasellus vel ante a risus rhoncus finibus ac sed nisl. Sed sapien lorem, gravida sit amet congue maximus, lobortis a sapien.',
     url: 'www.goto.com',
+    more: 'Read more',
```
Comment (fgravin) [inline libs/ui/src/lib/record-preview-list/record-preview-list.stories.ts:29]: to be removed

**Comment by MichelGabriel** [comment]
Yes, I will try to focus on just the dumb components. Problem was I didn't know where the dumb component stopped and the ES mapping began.

I'm going to try to split them into 3 components.

**Comment by MichelGabriel** [comment]
Replaced by https://github.com/geonetwork/geonetwork-ui/pull/25

**Pull request closed without merging**